crates/ruff_python_formatter/src/string/normalize.rs

```diff
@@ -913,11 +939,46 @@ pub(super) fn is_fstring_with_quoted_debug_expression(
         } else {
             false
         }
-    }) {
-        return true;
-    }
+    })
+}
 
-    false
+/// Returns `true` if `string` is an f-string part that contains a debug expression with a format spec
+/// that contains the opposite quote. It's important to preserve the quote style for those f-strings
+/// because changing the quote style would result in invalid syntax.
+///
+/// ```python
+/// f'{1=: "abcd \'\'}'
+/// ```
+pub(super) fn is_fstring_with_quoted_format_spec_and_debug(
```

Comment:
The documentation seems to be a bit misleading as per the logic. I don't see it considering the debug expression.

I would also reword it as "Returns `true` if the `f_string` has any part that ..."